packages/store/src/legacyMap.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+import { assert, details as X, q } from '@agoric/assert';
+
+/**
+ * @deprecated switch to ScalarMap if possible, Map otherwise
```

Comment:
maybe add a brief note about why you might be forced to use LegacyMap, and/or what would prevent you from switching to ScalarMap (using non-passable values?)

Reply:
I agree, I think it'd be helpful to gather up all the reasons why we're still using LegacyMap from the comments added in this PR, and use them as examples.

Reply:
The typical problem is indeed non-passable values.

I added I nice explanation to the doccomment. I didn't gather all the examples, but I used two clear examples and explained how to find the others. Resolving.